### PR TITLE
[MyCollection] Cleanup AddArtwork screen 

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Enhancement**

### Description

This PR does some final pre-backend cleanup for the Add Artwork screen:
- Matches comps
- Adds validation 

#skip_new_tests
#trivial

### CODE REVIEW

File: src/lib/Scenes/Consignments/v2/Screens/AddArtwork/Form/artworkSchema.tsx [hunk 1]
```diff
@@ -3,7 +3,11 @@ import { ArtworkFormValues } from "lib/Scenes/Consignments/v2/State/artworkModel
 import * as Yup from "yup"
 
 export const artworkSchema = Yup.object().shape({
-  artist: Yup.string().test("artist", "Artist must not be empty", value => value !== ""),
+  artistSearchResult: Yup.object()
+    .nullable()
+    .test("artistSearchResult", "Artist search result required", value => value !== null),
+  medium: Yup.string().test("medium", "Medium required", value => value !== ""),
+  size: Yup.string().test("size", "Size required", value => value !== ""),
```
Comment (damassi) [inline src/lib/Scenes/Consignments/v2/Screens/AddArtwork/Form/artworkSchema.tsx:10]: These validation messages aren't actually being used ATM, but they might be in the future. For now we're just enabling / disabling submit button and printing errors, in `__DEV__`, to the screen.

File: src/lib/Scenes/Consignments/v2/Screens/AddArtwork/MyCollectionAddArtwork.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ export const MyCollectionAddArtwork: React.FC = () => {
   const artworkState = useStoreState(state => state.artwork)
   const { formik } = useArtworkForm()
   const photos = artworkState.formValues.photos
+  const formattedTitleAndYear = [formik.values.title, formik.values.year].filter(Boolean).join(", ")
```
Comment (jonallured) [inline src/lib/Scenes/Consignments/v2/Screens/AddArtwork/MyCollectionAddArtwork.tsx:18]: I love `.filter(Boolean)` but wish it was just `compact`. 🤗

Comment (damassi) [reply src/lib/Scenes/Consignments/v2/Screens/AddArtwork/MyCollectionAddArtwork.tsx:18]: yah, so agree

**Comment by damassi** [comment]
Gonna merge!

**Comment by damassi** [comment]
#mergeOnGreen

**Pull request merged**